### PR TITLE
Docs [Fix]: use bytes instead of strings while writing

### PR DESCRIPTION
**Pull request by krshrimali**
## What does this PR do?

Sample examples given in [ENABLE ANY SERVER (BASIC)](https://lightning.ai/lightning-docs/workflows/add_server/any_server.html) page, fail with an error:

```python
File "/Users/krshrimali/Documents/Projects/GridAI/lightning/src/lightning_app/sample.py", line 11, in do_GET
    self.wfile.write(html)
  File "/opt/homebrew/Cellar/python@3.9/3.9.13_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/socketserver.py", line 826, in write
    self._sock.sendall(b)
TypeError: a bytes-like object is required, not 'str'
```

This needs to be fixed by passing bytes instead of strings. One of the solution would be to use `.encode()`, but usually, using `b"string"` is easy to convey that we want to convert to bytes before writing. Tested locally that using bytes makes the examples run successfully.

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @rohitgr7 @Felonious-Spellfire

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
Mind adding a note in the docs?

File: docs/source-app/workflows/add_server/any_server.rst [hunk 1]
```diff
@@ -31,7 +31,8 @@ Any server that listens on a port, can be enabled via a work. For example, here'
         def do_GET(self):
             self.send_response(HTTPStatus.OK)
             self.end_headers()
-            html = "<h1 style='color: blue'> Hello lit world </div>"
+            # Make sure to pass the data as bytes to the self.wfile.write call
```
Comment (krshrimali) [inline docs/source-app/workflows/add_server/any_server.rst:34]: @tchaton - Is this okay? I thought adding a comment will be better, as it's more about this line/example. Please let me know if it's not what you meant :)

File: docs/source-app/workflows/add_server/any_server.rst [hunk 2]
```diff
@@ -52,7 +53,8 @@ To enable the server inside the component, start the server in the run method an
         def do_GET(self):
             self.send_response(HTTPStatus.OK)
             self.end_headers()
-            html = "<h1 style='color: blue'> Hello lit world </div>"
+            # Make sure to pass the data as bytes to the `self.wfile.write` call
```
Comment (Felonious-Spellfire) [inline docs/source-app/workflows/add_server/any_server.rst:56]:
```suggestion
            # Data must pass as bytes to the `self.wfile.write` call
```

File: docs/source-app/workflows/add_server/any_server.rst [hunk 3]
```diff
@@ -81,7 +83,8 @@ In this case, we render the ``LitServer`` output in the ``home`` tab of the appl
...
```
Comment (Felonious-Spellfire) [inline docs/source-app/workflows/add_server/any_server.rst:86]:
```suggestion
            # Data must pass as bytes to the `self.wfile.write` call
```

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by Borda — APPROVED** [review]

**Pull request merged**